app/models/manageiq/providers/vmware/infra_manager/provision/configuration/container.rb

```diff
@@ -3,6 +3,12 @@ def build_config_spec
     VimHash.new("VirtualMachineConfigSpec") do |vmcs|
       vmcs.annotation = build_vm_notes
 
+      # Without these values a provision can't succeed, so default them to the vm name.
+      vm_name = get_option(:vm_name)
+      [:hostname, :linux_host_name, :vm_name, :vm_target_hostname, :vm_target_name].each do |option|
+        vmcs.send("#{option}=", options[option].presence || vm_name)
+      end
```

Comment:
@bdunne I'm confused why these values are being set in the `VirtualMachineConfigSpec` object.  None of these values are defined on that object [Data Object - VirtualMachineConfigSpec](http://pubs.vmware.com/vsphere-60/index.jsp?topic=%2Fcom.vmware.wssdk.apiref.doc%2Fvim.vm.ConfigSpec.html).  The example automate code referenced in the associated BZ shows setting these values in the provision task's options hash.

The hostname needs to be sanitized before passing to the provider. [app/models/miq_provision/helper.rb#L2](https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_provision/helper.rb#L2)  You would want to set these values before the hostname sanitized.  

Finally, it seems that you would want to set these values for other infra providers as well, not just vmware.